### PR TITLE
Refactoring + error handling fixes

### PR DESCRIPTION
**Pull request by jmigueprieto**
Follow up of https://github.com/conductor-sdk/conductor-go/pull/157

## Changes in this PR

1. Refactored http status code checking logic and fixed error handling in clients. 
2. Removed dead code.
3. Fixes in executor + executor tests.
4. Fixed tests.
5. Fixed `UserClient.GetGrantedPermissions()` implementation.

For more details take a look at the comments.

TODO: More testing (in a next PR)

### CODE REVIEW

File: sdk/authentication/token_helper.go [hunk 1]
```diff
@@ -73,24 +73,16 @@ func GetToken(credentials settings.AuthenticationSettings, httpSettings *setting
...
+		return localVarReturnValue, localVarHttpResponse, newErr
+	} else {
 		err = decode(&localVarReturnValue, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
 		if err == nil {
 			return localVarReturnValue, localVarHttpResponse, err
 		}
 	}
-	if localVarHttpResponse.StatusCode >= 300 {
-		newErr := fmt.Errorf(string(localVarBody))
-		if localVarHttpResponse.StatusCode == 200 {
```
Comment (jmigueprieto) [inline sdk/authentication/token_helper.go:84]: more dead code

File: test/integration_tests/executor_test.go [hunk 2]
```diff
@@ -1,25 +1,92 @@
...
 	if swaggerErr, ok := err.(client.GenericSwaggerError); ok {
-		// hmm... this should be a 404 or 400, but it's a 500 right now.
-		assert.Error(t, err, "GetWorkflow was expected to return a 500 error")
-		assert.Equal(t, 500, swaggerErr.StatusCode())
+		assert.Equal(t, 500, swaggerErr.StatusCode()) //TODO on test server update, should be a 404
+	} else {
+		assert.Fail(t, "err is not of type GenericSwaggerError")
+	}
+}
+
```
Comment (jmigueprieto) [inline test/integration_tests/executor_test.go:32]: Added some tests to check executor error responses

File: sdk/workflow/executor/executor.go [hunk 3]
```diff
@@ -426,7 +427,10 @@ func (e *WorkflowExecutor) UpdateTask(taskId string, workflowInstanceId string,
 		return err
 	}
 	taskResult.Status = status
-	e.taskClient.UpdateTask(context.Background(), taskResult)
+	_, _, err = e.taskClient.UpdateTask(context.Background(), taskResult)
+	if err != nil {
```
Comment (jmigueprieto) [inline sdk/workflow/executor/executor.go:431]: `executor.UpdateTask` was not handling errors properly

File: sdk/workflow/executor/executor.go [hunk 4]
```diff
@@ -83,17 +83,15 @@ func NewWorkflowExecutor(apiClient *client.APIClient) *WorkflowExecutor {
...
-	response, err := e.metadataClient.RegisterWorkflowDef(
+	_, err := e.metadataClient.RegisterWorkflowDef(
 		context.Background(),
 		overwrite,
 		*workflow,
 	)
 	if err != nil {
 		return err
 	}
-	if response.StatusCode > 299 {
```
Comment (jmigueprieto) [inline sdk/workflow/executor/executor.go:94]: These checks after 
```
if err != nil {
	return err
}
```

don't make sense. `err` is going to be not nil and this line won't be reached.

File: sdk/client/user_client.go [hunk 5]
```diff
@@ -2,14 +2,15 @@ package client
...
 	"net/http"
+
+	"github.com/conductor-sdk/conductor-go/sdk/model/rbac"
 )
 
 type UserClient interface {
 	CheckPermissions(ctx context.Context, userId string, type_ string, id string) (map[string]interface{}, *http.Response, error)
 	DeleteUser(ctx context.Context, id string) (*http.Response, error)
-	GetGrantedPermissions(ctx context.Context, userId string) ([]rbac.GrantedAccess, *http.Response, error)
+	GetGrantedPermissions(ctx context.Context, userId string) (rbac.GrantedAccessResponse, *http.Response, error)
```
Comment (jmigueprieto) [inline sdk/client/user_client.go:13]: The return type was wrong. Because of buggy error handling this was not caught by the tests before

**Review by c4lm — APPROVED** [review]

**Pull request merged**